### ISSUE TITLE
Encode requests using Grammars

### ISSUE THREAD

**Issue by rawleyfowler** [rawleyfowler/Humming-Bird#26]
Right now requests are encoded very naively, we can use grammars to make this better.

**Comment by masukomi** [comment]
Are you suggesting using a grammar to replace 

```
get('/', -> $request, $response {
    $response.html('<h1>Hello World</h1>');
});
```

so that it isn't something that LOOKS like a function body but is actually just a param passed to a function? 

Because if so I am 100% in support of this.

**Comment by rawleyfowler** [comment]
@masukomi 
Yeah I would like to make some sort of slang or grammar to do that

**Comment by rawleyfowler** [comment]
Obviously the current way will still have to exist to keep backwards compatibility.

**Comment by masukomi** [comment]
>  Obviously the current way will still have to exist to keep backwards compatibility.


this is early enough that i don't think it's bad to break backwards compatibility. a) there's hardly anyone using it b) it was pretty clear it was _very_ early stages c) they can just pin it to the old version if it matters to them.

I don't think it's worth maintaining tech debt indefinitely for such a small number of users and actual instances of the code in action.

**Comment by rawleyfowler** [comment]
I'm thinking that maybe the grammar can translate to the current scheme. We'll see though. Still pretty up in the air on how we should approach this.
